File: FeiDan 2/app/common/platform/Account.php
<?php

namespace app\common\platform;

use app\common\Http;
use Exception;

class Account extends Base
{
    /**
     * 登录-获取token
     * @param array $data
     * @return string
     * @throws Exception
     */
    public static function getToken(array $data = []): string
    {
        $response = Http::post(static::getApi('getToken'), [
            'url' => $data['url'],
            'username' => $data['username'],
            'password' => $data['password'],
        ]);
        $response = json_decode($response, true);
        if ($response['code'] !== 200) throw new Exception($response['msg'] ?? '服务器处理错误, 请重试');
        return $response['data']['token'];
    }

    public static function getUserInfo(array $data = []): array
    {
        $response = Http::post(static::getApi('getUserInfo'), [
            'url' => $data['url'],
            'token' => $data['token'],
        ]);
        $response = json_decode($response, true);
        if ($response['code'] !== 200) throw new Exception($response['msg'] ?? '服务器处理错误, 请重试');
        return $response['data'];
    }

    public static function getList(array $data = [])
    {
        $response = Http::post(static::getApi('getList'), [
            'url' => $data['url'],
            'token' => $data['token'],
        ]);
        $response = json_decode($response, true);
        if ($response['code'] !== 200) throw new Exception($response['msg'] ?? '服务器处理错误, 请重试');
        return $response['data'];
    }
}

File: feidan-api 2/api/v1/account/account.go
package account

import (
	"feidan-api/model/account/request"
	accountRes "feidan-api/model/account/response"
	"feidan-api/model/common/response"
	"github.com/gin-gonic/gin"
)

type AccountApi struct{}

func (s *AccountApi) GetToken(c *gin.Context) {
	var req request.GetTokenReq
	if err := c.ShouldBind(&req); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	token, err := apiService.AccountService.GetToken(c, req.Url, req.Username, req.Password)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.OkWithDetailed(&accountRes.GetTokenRsp{Token: token}, "登陆成功", c)
}

func (s *AccountApi) GetUserInfo(c *gin.Context) {
	var req request.GetUserInfoReq
	if err := c.ShouldBind(&req); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	info, err := apiService.AccountService.GetUserInfo(c, req.Url, req.Token)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.OkWithDetailed(info, "获取用户信息成功", c)
}

func (s *AccountApi) GetList(c *gin.Context) {
	var req request.GetListReq
	if err := c.ShouldBind(&req); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	list, err := apiService.AccountService.GetList(c, req.Url, req.Token)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.OkWithDetailed(list, "获取成功", c)
}

func (s *AccountApi) AutoLogin(c *gin.Context) {
	result, err := apiService.AccountService.AutoLogin(c)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.HtmlString(result, c)
}
